pkg/cmd/roachtest/sqlalchemy.go

```diff
@@ -0,0 +1,359 @@
+			scanner := bufio.NewScanner(bytes.NewReader(rawResults))
+			for scanner.Scan() {
+				match := sqlAlchemyResultRegex.FindStringSubmatch(scanner.Text())
+				if match != nil {
+					groups := map[string]string{}
+					for i, name := range match {
+						groups[sqlAlchemyResultRegex.SubexpNames()[i]] = name
+					}
+					// There are duplicate tests that match the passed in subset. So
+					// apened the test set letter to the end of the test.
+					test := fmt.Sprintf("%s--%s", groups["name"], testset)
+					pass := groups["result"] == "PASSED"
+
+					// Check for duplicates within the set and add a number. Luckily,
+					// all tests are preformed in the same order on each run.
+					x := 0
+					for {
+						newTestName := fmt.Sprintf("%s:%d", test, x)
+						if _, ok := runTests[newTestName]; !ok {
+							break
+						}
+						x++
+					}
+					test = fmt.Sprintf("%s:%d", test, x)
+
+					runTests[test] = struct{}{}
+					allTests = append(allTests, test)
+
+					ignoredIssue, expectedIgnored := ignoredlist[test]
+					issue, expectedFailure := expectedFailureList[test]
+					switch {
+					case expectedIgnored:
+						results[test] = fmt.Sprintf("--- SKIP: %s due to %s (expected)", test, ignoredIssue)
+						ignoredCount++
+					case pass && !expectedFailure:
+						results[test] = fmt.Sprintf("--- PASS: %s (expected)", test)
+						passExpectedCount++
+					case pass && expectedFailure:
+						results[test] = fmt.Sprintf("--- PASS: %s - %s (unexpected)",
+							test, maybeAddGithubLink(issue),
+						)
+						passUnexpectedCount++
+					case !pass && expectedFailure:
+						results[test] = fmt.Sprintf("--- FAIL: %s - %s (expected)",
+							test, maybeAddGithubLink(issue),
+						)
+						failExpectedCount++
+						currentFailures = append(currentFailures, test)
+					case !pass && !expectedFailure:
+						results[test] = fmt.Sprintf("--- FAIL: %s (unexpected)", test)
+						failUnexpectedCount++
+						currentFailures = append(currentFailures, test)
+					}
+				}
+			}
+		}
+
+		runTestsets := func(nodeID int) {
+			defer runWG.Done()
+			node := c.Node(nodeID)
+
+			for testset := getNextTestset(); len(testset) > 0; testset = getNextTestset() {
+				c.l.Printf("n%d - running SQL Alchemy test suite for tests %s", nodeID, testset)
+				rawResults, _ := c.RunWithBuffer(ctx, t.l, node, runTestCommand(testset))
+				c.l.Printf("Test Results for tests %s:\n%s", testset, rawResults)
+
+				if t.Failed() || ctx.Err() != nil {
+					return
+				}
+
+				// Lock the processing of results since the tallying is not thread safe.
+				c.l.Printf("n%d - collating the test results for tests %s", nodeID, testset)
+				collateResults(testset, rawResults)
+				c.l.Printf("n%d - finished processing tests %s", nodeID, testset)
+				c.l.Printf("-- Remaining test sets: %v", remainingTestsets)
+			}
+
+			c.l.Printf("n%d - no more tests to run, exiting", nodeID)
+		}
+
+		// Note that this is expected to return an error, since the test suite
```

Comment:
What exactly is expected to return an error? runTestsets doesn't return anything.